src/ast.rs

```diff
@@ -72,50 +73,47 @@ impl From<((usize, usize, usize), (usize, usize, usize))> for Span {
 }
 
 /// Represents a name of an entity, such as a type, variable, function, etc.
-#[derive(Debug, PartialEq)]
-pub struct RawIdentifier<'a> {
-    pub name: &'a str,
-}
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawIdentifier(pub String);
 
-pub type Identifier<'a> = Spanned<RawIdentifier<'a>>;
+pub type Identifier = Spanned<RawIdentifier>;
 
-impl<'a> From<ParseInput<'a>> for RawIdentifier<'a> {
+impl<'a> From<ParseInput<'a>> for RawIdentifier {
     fn from(span: ParseInput<'a>) -> Self {
-        RawIdentifier {
-            name: span.fragment(),
-        }
+        RawIdentifier((*span.fragment()).to_owned())
     }
 }
 
-/// A parameter to a function, e.g., `foo: MyType`.
-#[derive(Debug, PartialEq)]
-pub struct RawParameter<'a> {
-    pub name: Identifier<'a>,
-    pub param_type: Identifier<'a>,
+/// Introduces a variable with its name and type, e.g., `foo : MyType`. E.g. used in function
+/// type signature parameter lists, `let` bindings, etc.
+#[derive(Debug, PartialEq, Clone)]
+pub struct RawVariableDeclaration {
+    pub name: Identifier,
+    pub typ: Identifier,
 }
 
-pub type Parameter<'a> = Spanned<RawParameter<'a>>;
-pub type ParameterList<'a> = Spanned<Vec<Parameter<'a>>>;
+pub type VariableDeclaration = Spanned<RawVariableDeclaration>;
+pub type VariableDeclarationList = Spanned<Vec<VariableDeclaration>>;
 
-impl<'a> From<(Identifier<'a>, Identifier<'a>)> for RawParameter<'a> {
-    fn from((name, param_type): (Identifier<'a>, Identifier<'a>)) -> Self {
-        RawParameter { name, param_type }
+impl From<(Identifier, Identifier)> for RawVariableDeclaration {
+    fn from((name, typ): (Identifier, Identifier)) -> Self {
```

Comment:
We really gotta get TypeDef and TypeRef going soon, b/c Identifier-as-type is a sad hack

(Note to self: contribute)

Reply:
For the record, during parsing, I would not say that this is a hack. During parsing, there is no such thing as TypeDef or TypeRef right? We can't determine these things until we start doing more semantic analysis of the parsed result (e.g. type checking). Right?

Reply:
It _is_ the case that types must be declared in usage order...I think this warrants a chat w/Leary to see if he has Design Opinions on how this evolves. In any case, this is nothing to block this PR. Just make some record somewhere (TODO, issue, whatever) so we don't forget to follow up.